### PR TITLE
fix: Generate model files before analyzing future calls

### PR DESCRIPTION
**Pull request by nielsenko**
When no generated files exist on disk (first checkout, generated folder deleted, etc.), `serverpod generate` could fail on first run. You could often fix by running again, but there was a corner case regarding auto-generated parameter models for future calls.

This PR fixes both.

Also removes `DartTemporaryProtocolGenerator` and related tests, as it is no longer used.

Fixes ? 

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [X] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None.

### CODE REVIEW

**Comment by nielsenko** [comment]
@coderabbit review

File: tools/serverpod_cli/lib/src/util/analysis_helpers.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/// modified on disk, so subsequent analysis calls resolve updated content.
+Future<void> refreshAnalysisContext(
+  AnalysisContextCollection collection,
+  Iterable<String> filePaths,
+) async {
+  final context = collection.contexts.single; // current invariant
+  for (final path in filePaths) {
+    context.changeFile(p.canonicalize(path));
+  }
+  await context.applyPendingFileChanges();
```
[the remark this answers is not in the document]
Comment (nielsenko) [reply tools/serverpod_cli/lib/src/util/analysis_helpers.dart:19]: Did you read `// current invariant`? I prefer this to blow up if the invariant is broken, rather than try to handle it with a loop.

**Comment by nielsenko** [comment]
Cleaned up history for human review.

**Comment by nielsenko** [comment]
rebased on main
